src/classes/etoken.ts

```diff
@@ -183,20 +215,35 @@ export class Etoken {
         }
     }
 
-    public login(token: number, password: string) {
+    public login(token: number, password: string): boolean {
         try {
             this.initSession(token);
             if (!this.session) throw new Error('The token session is not already yet');
             this.PKCS11.C_Login(this.session, 1, password);
             this.tokenInUse = this.getTokenConnectedInfo(token);
+            this.loggedIn = true;
             return true;
         } catch(e) {
             this.finalize();
             return false;
         }
     }
 
-    public verifyTokenInUse() {
+    public loginPersistent(token: number, password: string): boolean {
+        try {
+            this.initSession(token);
+            if (!this.session) throw new Error('The token session is not already yet');
+            this.PKCS11.C_Login(this.session, 1, password);
+            this.tokenInUse = this.getTokenConnectedInfo(token);
+            this.loggedIn = true;
+            return true;
+        } catch(e) {
+            console.log(e);
```

Comment:
Atento aqui deberia finalizar la sesión. Sino el token queda abierto en la inicialización del token
deberias usar this.finalize();

Reply:
Si, en el catch, aunque los métodos son iguales quizás deberíamos dejar solo uno.